services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/IgnoreNullsMode.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb.model;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+
+/**
+ * This enum offers users different modes of performing DDB item updates
+ * <p>
+ * In the SCALAR_ONLY mode, updates to nested scalar attributes are supported
+ * <p>
+ * In the MAPS_ONLY mode, updates to nested map structures are supported
+ * <p>
+ * The DEFAULT mode operates by setting ignoreNulls to false, and requires the user to
+ * fetch existing DDB item, make modifications to it and then update the item
+ */
+@SdkPublicApi
+public enum IgnoreNullsMode {
+    SCALAR_ONLY,
+    MAPS_ONLY,
+    DEFAULT
```

Comment:
What is the difference between MAPS_ONLY and DEFAULT? In the code, seems like we behave the same unless its `SCALAR_ONLY`

Reply:
I have modified this so that we do things differently. Basically, 

SCALAR_ONLY - when user wants to update nested attributes while only providing the changeset, i.e. today's ignoreNulls = true mode. 

MAPS_ONLY - This is a weird case that works the same for when ignoreNulls = true / false, i.e. when user provides only the new data, or all the data. The reason it works is because this mode ONLY supports updating with new maps, so the delta of changes constitutes the entire map. 

DEFAULT - when the user provides the whole item with the modified data. equivalent to ignoreNulls = false. 

We only need query transformations when the user sets SCALAR_ONLY mode and not otherwise. I had the modes defined jsut to make it clear. I will discuss a bit to see if we can remove it